app/controllers/cart_controller.rb

```diff
@@ -17,8 +17,12 @@ def add_quantity
   def decrease_quantity
     item = Item.find(params[:item_id])
     quantity = params[:quantity_decrease].to_i
-      cart.remove_quantity(quantity, item)
-      redirect_to '/cart'
+      if cart.remove_quantity(quantity, item) != false
+        redirect_to '/cart'
+      else
+        session[:cart].delete(params[:item_id])
+        redirect_to '/cart'
```

Comment:
This is so when the quantity is decremented to below 0 it will delete out of the cart.

Reply:
Nice Work